src/steam_network/protocol/protobuf_client.py

```diff
@@ -168,77 +321,211 @@ async def _get_obfuscated_private_ip(self) -> int:
+        """
+            var logon = new ClientMsgProtobuf<CMsgClientLogon>( EMsg.ClientLogon );
+
+            SteamID steamId = new SteamID( details.AccountID, details.AccountInstance, Client.Universe, EAccountType.Individual );
+
+            if ( details.LoginID.HasValue )
+            {
+                // TODO: Support IPv6 login ids?
+                logon.Body.obfuscated_private_ip = new CMsgIPAddress
+                {
+                    v4 = details.LoginID.Value
+                };
+            }
+            else
+            {
+                logon.Body.obfuscated_private_ip = NetHelpers.GetMsgIPAddress( this.Client.LocalIP! ).ObfuscatePrivateIP();
+            }
+
+            // Legacy field, Steam client still sets it
+            if ( logon.Body.obfuscated_private_ip.ShouldSerializev4() )
+            {
+                logon.Body.deprecated_obfustucated_private_ip = logon.Body.obfuscated_private_ip.v4;
+            }
+
+            logon.ProtoHeader.client_sessionid = 0;
+            logon.ProtoHeader.steamid = steamId.ConvertToUInt64();
+
+            logon.Body.account_name = details.Username; //Null
+            logon.Body.password = details.Password; //Null
+            logon.Body.should_remember_password = details.ShouldRememberPassword; //false
+
+            logon.Body.protocol_version = MsgClientLogon.CurrentProtocol; 
+            logon.Body.client_os_type = ( uint )details.ClientOSType; //
+            logon.Body.client_language = details.ClientLanguage;
+            logon.Body.cell_id = details.CellID ?? Client.Configuration.CellID //
+
+            logon.Body.steam2_ticket_request = details.RequestSteam2Ticket;
+
+            // we're now using the latest steamclient package version, this is required to get a proper sentry file for steam guard
+            logon.Body.client_package_version = 1771; // todo: determine if this is still required
+            logon.Body.supports_rate_limit_response = true;
+            logon.Body.machine_name = details.MachineName;
+            logon.Body.machine_id = HardwareUtils.GetMachineID( Client.Configuration.MachineInfoProvider );
+
+            // steam guard 
+            logon.Body.auth_code = details.AuthCode;
+            logon.Body.two_factor_code = details.TwoFactorCode;
+
+#pragma warning disable CS0618 // LoginKey is obsolete
+            logon.Body.login_key = details.LoginKey;
+#pragma warning restore CS0618 // LoginKey is obsolete
+
+            logon.Body.access_token = details.AccessToken;
+
+            logon.Body.sha_sentryfile = details.SentryFileHash;
+            logon.Body.eresult_sentryfile = ( int )( details.SentryFileHash != null ? EResult.OK : EResult.FileNotFound );
+
+
+            this.Client.Send( logon );
+        }
+        """
+        resetSteamIDAfterThisCall : bool = False
+        if (self.confirmed_steam_id is None):
+            resetSteamIDAfterThisCall = True
+            self.confirmed_steam_id = steam_id
```

Comment:
it actually does, which was maddening. 

We call _send here
https://github.com/GOG-Nebula/galaxy-integration-steam/pull/1/files/080a8d2ae57a4379a680ef371cabf2a4d6b43b1c#diff-c14b135777af3a9c59f474969ec8597ad639b3891330600c1fa508eecbed2210R417

and it's used in send here.
https://github.com/GOG-Nebula/galaxy-integration-steam/pull/1/files/080a8d2ae57a4379a680ef371cabf2a4d6b43b1c#diff-c14b135777af3a9c59f474969ec8597ad639b3891330600c1fa508eecbed2210R546-R550

this is an incredibly hacky fix.

Reply:
those link does not work for me :/ - could you please make links from commit instead of from PR diff?

Reply:
updated the links to point to my master branch (which this PR is based on). Those links should be better.

Reply:
Ah, ok it makes sense now. I see the problem.

It could feel less hacky if we add `override_steam_id` parameter to `_send` -- then `resetSteamIDAfterThisCall` var is not needed. Another option to increase redability is to use explicite contextmanager that flips self.confirmed_steam_id. But I would chose the first option, as such contextmanager is too powerful so someone could be temped to use it around awaitable code and then random messages could have been sent with different steam_id

Reply:
I think i actually have the override version in a dev branch, let me check. I just used an optional which defaults to null. If it isn't null, use that instead. It doesn't affect any other functions since it's an optional parameter.